qos-monitor/src/main/java/eu/arrowhead/core/qos/service/ping/monitor/impl/OrchestratedExternalPingMonitor.java

```diff
@@ -0,0 +1,270 @@
+		case NOT_SECURE:
+			return createHttpUri(provider, path);
+		case CERTIFICATE:
+			return createHttpsUri(provider, path);
+		case TOKEN:
+			throw new InvalidParameterException("TOKEN security is not supported yet, for PingMonitor Providers. ");
+		default:
+			throw new InvalidParameterException("Not supported ServiceSecurityType: " + securityType);
+		}
+
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	private UriComponents createHttpUri(final SystemResponseDTO provider, final String path) {
+
+		return Utilities.createURI(CommonConstants.HTTP, provider.getAddress(), provider.getPort(), path );
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	private UriComponents createHttpsUri(final SystemResponseDTO provider, final String path) {
+
+		return Utilities.createURI(CommonConstants.HTTPS, provider.getAddress(), provider.getPort(), path );
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	private int calculateTimeOut() {
+
+		final int singlePingTimeOut = pingMeasurementProperties.getTimeout();
+		final int timesToRepeatPing = pingMeasurementProperties.getTimeToRepeat();
+
+		return singlePingTimeOut * timesToRepeatPing * OVERHEAD_MULTIPLIER;
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	private SystemRequestDTO getPingMonitorSystemRequestDTO() {
+		logger.debug("getPingMonitorSystemRequestDTO started...");
+
+		Assert.notNull(cachedPingMonitorProvider.getProvider(), "Cached Provider is null");
+
+		final SystemResponseDTO provider = cachedPingMonitorProvider.getProvider();
+
+		final SystemRequestDTO system = new SystemRequestDTO();
+		system.setSystemName(provider.getSystemName());
+		system.setAddress(provider.getAddress());
+		system.setPort(provider.getPort());
+		system.setMetadata(provider.getMetadata());
+		system.setAuthenticationInfo(provider.getAuthenticationInfo());
+
+		return system;
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	private void initPingMonitorProvider() {
+		logger.debug("initPingMonitorProvider started...");
+
+		final OrchestrationFormRequestDTO request = orchestrationRequestFactory.createExternalMonitorOrchestrationRequest();
+		final OrchestrationResponseDTO result = driver.queryOrchestrator(request);
+
+		cachedPingMonitorProvider = selectProvider(result);
+
+		driver.unsubscribeFromPingMonitorEvents();
+		driver.subscribeToExternalPingMonitorEvents(getPingMonitorSystemRequestDTO());
+
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	private OrchestrationResultDTO selectProvider(final OrchestrationResponseDTO result) {
+		logger.debug("selectProvider started...");
+
+		//TODO implement more sophisticated provider selection strategy
```

Comment:
and you should be able to handle the empty response too

Reply:
fixed